### ISSUE TITLE
build pangolin-desktop error.

### ISSUE THREAD

**Issue by liudonghua123** [dahliaOS/backend#4]
I tried to build an android apk of `pangolin-desktop`, and I got the following errors.

```shell
Error: unable to find directory entry in pubspec.yaml: D:\apps\flutter_master\.pub-cache\git\backend-3d3529768add379b82e2cde8b909228e494dba86\..\global\assets\
This application cannot tree shake icons fonts. It has non-constant instances of IconData at the following locations:
  - file:///D:/code/flutter/pangolin-desktop/lib/desktop/taskbar/launcher.dart:92:35


FAILURE: Build failed with an exception.
```

It seems lack of `../global/assets/` resources.

https://github.com/dahliaOS/backend/blob/3d3529768add379b82e2cde8b909228e494dba86/pubspec.yaml#L31-L33

**Comment by liudonghua123** [comment]
If I comment these three lines, I got another errors.

```shell
This application cannot tree shake icons fonts. It has non-constant instances of IconData at the following locations:
  - file:///D:/code/flutter/pangolin-desktop/lib/desktop/taskbar/launcher.dart:92:35


FAILURE: Build failed with an exception.
```

See the related code belows.

https://github.com/dahliaOS/pangolin_desktop/blob/71a8bbb50ee71b0371e829802ecde5c59c1af189/lib/desktop/taskbar/launcher.dart#L91-L93

**Comment by larsb24** [comment]
Guessing you're on windows, use powershell and run this command: ```git submodule update --init --recursive && cd backend && git checkout main && cd ..```

**Comment by liudonghua123** [comment]
@larsb24 Thanks, for this error, I can fix it via execute the build with a `--no-tree-shake-icons` option, so finally, I can build successful via `flutter build apk --no-tree-shake-icons`.

> If I comment these three lines, I got another errors.
> 
> ```shell
> This application cannot tree shake icons fonts. It has non-constant instances of IconData at the following locations:
>   - file:///D:/code/flutter/pangolin-desktop/lib/desktop/taskbar/launcher.dart:92:35
> 
> 
> FAILURE: Build failed with an exception.
> ```
> 
> See the related code belows.
> 
> https://github.com/dahliaOS/pangolin_desktop/blob/71a8bbb50ee71b0371e829802ecde5c59c1af189/lib/desktop/taskbar/launcher.dart#L91-L93

**Comment by liudonghua123** [comment]
@larsb24 Thanks, I did not notice `pangolin_desktop` is a submodule repo, and I changed the `dahlia_backend` to use git url in pubspec.yaml previously. Now I tried to execute `git submodule update --init --recursive && cd backend && git checkout main && cd ..` and fix https://github.com/dahliaOS/welcome/issues/1 locally, and it works now.

**Comment by larsb24** [comment]
> @larsb24 Thanks, I did not notice `pangolin_desktop` is a submodule repo, and I changed the `dahlia_backend` to use git url in pubspec.yaml previously. Now I tried to execute `git submodule update --init --recursive && cd backend && git checkout main && cd ..` and fix [dahliaOS/welcome#1](https://github.com/dahliaOS/welcome/issues/1) locally, and it works now.

@liudonghua123 so can we close this error?

**Comment by liudonghua123** [comment]
@larsb24 Thanks again. 😄
